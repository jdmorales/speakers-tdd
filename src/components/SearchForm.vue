  <template>
  <div>
    <input type="text" v-model="inputData">
    <button @click="fetchSpeaker(inputData) "></button>
  </div>
</template>

<script>
import { mapActions } from 'vuex'

export default {
  data() {
    return {
      inputData: ''
    }
  },
  methods: {
    ...mapActions({
      fetchSpeaker: 'FETCH_SPEAKER'
    })
  }
}
</script>

<style>

</style>