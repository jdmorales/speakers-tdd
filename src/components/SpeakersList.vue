<template>
  <div>
    <Speaker v-for="speaker in speakers" :key="speaker.username"/>
  </div>
</template>

<script>
import Speaker from '@/components/Speaker'
export default {
  components: {
    Speaker
  },
  props: {
    speakers: {
      type: Array,
      required: true
    }
  }
}
</script>

<style>

</style>