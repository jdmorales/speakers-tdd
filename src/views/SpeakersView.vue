<template>
  <div>
    <SearchForm/>
    <SpeakersList :speakers="speakers"/>
  </div>
</template>

<script>
import SearchForm from '@/components/SearchForm'
import SpeakersList from '@/components/SpeakersList'
import { mapGetters } from 'vuex'

export default {
  name: 'SpeakersView',
  components: {
    SearchForm,
    SpeakersList
  },
  computed: {
    ...mapGetters({speakers: 'speakers'})
  }
}
</script>