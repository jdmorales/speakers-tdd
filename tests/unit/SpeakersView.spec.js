import { shallowMount } from '@vue/test-utils'
import SpeakersView from '@/views/SpeakersView'
import SearchForm from '@/components/SearchForm'
import SpeakersList from '@/components/SpeakersList'


describe('SpeakersView', () => {
  const build = () => {
    const wrapper = shallowMount(SpeakersView)

    return {
      wrapper,
      SearchForm: () => wrapper.find(SearchForm),
      SpeakersList: () => wrapper.find(SpeakersList)
    }
  }

  it('it renders the component', () => {
    // arrange
    const wrapper = shallowMount(SpeakersView)

    // assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('it renders the right', () => {
    // arrange
    const { SearchForm, SpeakersList } = build()

    // assert
    expect(SearchForm().exists()).toBe(true)
    expect(SpeakersList().exists()).toBe(true)
  })
})
