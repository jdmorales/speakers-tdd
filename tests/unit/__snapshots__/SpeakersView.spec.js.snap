// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeakersView renders the component 1`] = `
<div class="container">
  <searchform-stub></searchform-stub>
  <speakerslist-stub speakers="[object Object],[object Object]"></speakerslist-stub>
</div>
`;
